Sources/ReactiveMoya/MoyaProvider+Reactive.swift

```diff
@@ -29,8 +29,7 @@ public extension Reactive where Base: MoyaProviderType {
 
     /// Designated request-making method with progress.
     func requestWithProgress(_ token: Base.Target, callbackQueue: DispatchQueue? = nil) -> SignalProducer<ProgressResponse, MoyaError> {
-        let progressBlock: (Signal<ProgressResponse, MoyaError>.Observer) -> (ProgressResponse) -> Void = { observer in
-            return { progress in
+        let progressBlock: (Signal<ProgressResponse, MoyaError>.Observer) -> (ProgressResponse) -> Void = { observer in { progress in
```

Comment:
Here is a place where maybe its harder to read when omit the return...

Reply:
I think we should keep the return when the return type is a closure.
It is a cool feature when we want to have pretty compact computed properties (like when declaring a static TargetType) , but can quickly become uncomfortable in other use cases.

Reply:
Can we write `progressBlock` in shorter form ?

`let progressBlock = Signal<ProgressResponse, MoyaError>.Observer.send(value:)`

Reply:
> I think we should keep the return when the return type is a closure.
> It is a cool feature when we want to have pretty compact computed properties (like when declaring a static TargetType), but can quickly become uncomfortable in other use cases.

Yeah @amaurydavid, I totally agree :)) I do think we could make convention where omit the return type is only allowed when the single line could be a one-liner (which is the case for most common case for computed properties with single expression) other cases, as you said we can keep explicit

Reply:
> Can we write `progressBlock` in shorter form ?
> 
> `let progressBlock = Signal<ProgressResponse, MoyaError>.Observer.send(value:)`

Not sure, I will check ... thank you :)

Reply:
yes, imo let's just try to be reasonable and see if omitting the return keyword will help or not and decide base on that - in this case the shorter form proposed by @hamdshah seems to be working a lot better though!